demo/src/app/shared/component-wrapper/component-wrapper.component.ts

```diff
@@ -84,37 +84,39 @@ export class ComponentWrapper implements OnDestroy {
 	}
 
 	updateNavigation(component: NgbdExamplesPage | NgbdApiPage | any) {
-		const getLinks = (typeCollection: string[]) => {
-			return typeCollection.map((item) => ({
-				fragment: item,
-				title: item,
-			}));
-		};
-		this.tableOfContents = [];
-		if (component instanceof NgbdExamplesPage) {
-			this.tableOfContents = component.demos.map((demo) => {
-				return {
-					fragment: demo.id,
-					title: demo.title,
-				};
-			});
-		} else if (component instanceof NgbdApiPage) {
-			let toc = getLinks(component.components);
-
-			if (component.classes.length > 0) {
-				const klasses = getLinks(component.classes);
-				toc = toc.concat(toc.length > 0 ? [<any>{}, ...klasses] : klasses);
+		setTimeout(() => {
```

Comment:
Just curious, but why is the setTimeout needed here?

Reply:
I have not digged too much, but I'm pretty sure it's because of this change: https://angular.io/guide/update-to-version-15#v15-bc-09

This causes a NG0100 error, because this method is called when the router is activated and causes new changes.

Reply:
I did a `queueMicrotask()` in #4428, but haven't checked the profile on what is better to do